src/metrics/rules/active_ruleset.go

```diff
@@ -845,6 +858,17 @@ func (as *activeRuleSet) cutoverNanosAt(idx int) int64 {
 	return timeNanosMax
 }
 
+// isIncludeTag checks if tagName is in includeTagNames
+// Include tags are tags that if present in the metric id, should be included in the rollup id.
+func isIncludeTag(tagName []byte, includeTagNames [][]byte) bool {
+	for _, includeTagName := range includeTagNames {
```

Comment:
Scanning the array of `includeTagNames` upon every tag from original metric is inefficient. Store the `includeTagNames` as a map and simply check if given name is present in the map. That'd be O(1) operation instead of O(N).

Reply:
Feng left a similar [comment](https://github.com/m3db/m3/pull/4270#discussion_r1626350375). We would need to convert []byte to a string when looking up the map because keys of a map can't be []byte. In this case is changing to a map still faster?

Reply:
I made `includeTagNames` a map with keys of type uint64 and used murmur3 hashing library to hash the tagName into int64 to lookup in the map. This bypasses the []byte to string conversion, and uses a map for fast O(1) lookup